Packs/GoogleCloudFunctions/Integrations/GoogleCloudFunctions/GoogleCloudFunctions.py

```diff
@@ -0,0 +1,177 @@
+
+    # disable-secrets-detection-start
+    @staticmethod
+    def get_http_client_with_proxy():
+        proxies = handle_proxy()
+        if not proxies or not proxies['https']:
+            raise Exception('https proxy value is empty. Check Demisto server configuration')
+        https_proxy = proxies['https']
+        if not https_proxy.startswith('https') and not https_proxy.startswith('http'):
+            https_proxy = 'https://' + https_proxy
+        parsed_proxy = urllib.parse.urlparse(https_proxy)
+        proxy_info = httplib2.ProxyInfo(
+            proxy_type=httplib2.socks.PROXY_TYPE_HTTP,
+            proxy_host=parsed_proxy.hostname,
+            proxy_port=parsed_proxy.port,
+            proxy_user=parsed_proxy.username,
+            proxy_pass=parsed_proxy.password)
+        return httplib2.Http(proxy_info=proxy_info)
+
+    # disable-secrets-detection-end
+
+    def functions_list(self, region=None, project_id=None):
+        if project_id:
+            self.project = project_id
+        if region:
+            self.region = region
+        parent = f'projects/{self.project}/locations/{self.region}'
+        return self.service.projects().locations().functions().list(parent=parent).execute()
+
+    def region_list(self, project_id):
+        if project_id:
+            self.project = project_id
+        name = f'projects/{self.project}'
+        return self.service.projects().locations().list(name=name).execute()
+
+    def function_by_name(self, function_name, region=None, project_id=None):
+        if project_id:
+            self.project = project_id
+        if region:
+            self.region = region
+        name = f'projects/{self.project}/locations/{self.region}/functions/{function_name}'
+        return self.service.projects().locations().functions().get(name=name).execute()
+
+    def execute_function(self, function_name: str, data: str, region, project_id):
+        if project_id:
+            self.project = project_id
+        if region:
+            self.region = region
+        name = f'projects/{self.project}/locations/{self.region}/functions/{function_name}'
+        body = {'data': data}
+        return self.service.projects().locations().functions().call(name=name, body=body).execute()
+
+
+'''COMMAND FUNCTIONS'''
+
+
+def functions_list_command(client: GoogleClient, args: dict):
+    region = args.get('region')
+    project_id = args.get('project_id')
+    res = client.functions_list(region, project_id)
+    functions = res.get('functions', [])
+    keys = list(functions[0].keys())
+    keys.remove('name')
+    disp_region = 'All' if client.region == '-' else client.region
+    hr = tableToMarkdown(f'Functions in project "{client.project}" and region "{disp_region}"',
+                         functions, ['name'] + keys)
+    ec = {'GoogleCloudFunctions.Function(val.name && val.name == obj.name)': functions}
```

Comment:
is a function name unique across all regions/projects? if not, add another key to the DT

Reply:
yes. The function's name is its full hierarchy: projects/gcp-integrations/locations/us-central1/functions/arseny-func